File: swift/[M]371.Sum-of-Two-Integers.swift
/**
 * Runtime: 0 ms, faster than 100.00% of Swift online submissions for Sum of Two Integers.
 * Memory Usage: 13.9 MB, less than 37.84% of Swift online submissions for Sum of Two Integers.
 */
class Solution {
    func getSum(_ a: Int, _ b: Int) -> Int {
        var additionValue = b
        var result = a
        while additionValue != 0 {
            let carry = result & additionValue
            result = result ^ additionValue
            additionValue = carry << 1 
        }
        return result
    }
}

File: python/[M]371.Sum-of-Two-Integers.py
# Runtime: 28 ms, faster than 79.33% of Python3 online submissions for Sum of Two Integers.
# Memory Usage: 14 MB, less than 98.26% of Python3 online submissions for Sum of Two Integers.
class Solution:
    def getSum(self, a: int, b: int) -> int:
        MASK = 0xFFFFFFFF
        MAXVAL = 0x7FFFFFFF
        additionValue = b
        result = a
        while additionValue != 0:
            carry = result & additionValue
            result = (result ^ additionValue) & MASK
            additionValue = (carry << 1) & MASK
            
        if result > MAXVAL:
            result = ~(result ^ MASK)
            
        return result
